bolt_windows.go

```diff
@@ -67,6 +67,20 @@ func mmap(db *DB, sz int) error {
 	var sizelo, sizehi uint32
 
 	if !db.readOnly {
+		if db.MaxSize != 0 && db.MaxSize < sz {
+			// We need to truncate the file to memory map it on windows
+			// Make sure we are not going to grow the file
+			fileSize, err := db.fileSize()
+			if err != nil {
+				return fmt.Errorf("could not check existing db file size: %s", err)
+			}
+
+			if fileSize < sz {
+				// We can't mmap without growing the file
+				return errors.ErrMaxSizeReached
+			}
+		}
+
 		// Truncate the database to the size of the mmap.
 		if err := db.file.Truncate(int64(sz)); err != nil {
```

Comment:
We only need to truncate the file iff `fileSize < sz` is `true`.

Reply:
I think you're referring to the comment on line 71, which I've adjusted. This specific line, line 80, is indeed in the conditional of `fileSize < sz`

Reply:
oh, I think you're saying that the actual truncate command should be inside that conditional.

I considered this, but I wanted to avoid the file stat unless the sz requested was above the max size - otherwise we'd need to stat the file every time we mmap which I was afraid would have performance impact. Therefore, I don't have access to fileSize here

Reply:
this retains the existing behaviour when MaxSize is not set

Reply:
OK, let's just keep the existing behaviour (always truncating the file regardless)